framework/components/blockchain/canton/canton.go

```diff
@@ -0,0 +1,362 @@
+package canton
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+)
+
+// Canton Defaults
+const (
+	SpliceVersion = "0.5.3"
+	CantonImage   = "ghcr.io/digital-asset/decentralized-canton-sync/docker/canton"
+)
+
+// JWT Auth defaults
+const (
+	DefaultAuthProviderAudience = "https://chain.link"
+	DefaultUserName             = "api-user"
+)
+
+// Port prefixes for participants
+const (
+	DefaultParticipantJsonApiPortPrefix      = "11"
+	DefaultParticipantAdminApiPortPrefix     = "12"
+	DefaultLedgerApiPortPrefix               = "13"
+	DefaultHTTPHealthcheckPortPrefix         = "15"
+	DefaultGRPCHealthcheckPortPrefix         = "16"
+	DefaultSpliceValidatorAdminApiPortPrefix = "22"
+)
+
+func getCantonHealthCheckScript(numberOfValidators int) string {
+	script := `
+#!/bin/bash
+# Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -eou pipefail
+
+# SV
+echo "Checking ${CANTON_PARTICIPANT_GRPC_HEALTHCHECK_PORT_PREFIX}00"
+grpcurl -plaintext "localhost:${CANTON_PARTICIPANT_GRPC_HEALTHCHECK_PORT_PREFIX}00" grpc.health.v1.Health/Check
+
+	`
+	// Add additional participants
+	for i := range numberOfValidators {
+		i += 1 // start from 1 since SV is 0
+		script += fmt.Sprintf(`
+# Participant %02[1]d
+echo "Checking ${CANTON_PARTICIPANT_GRPC_HEALTHCHECK_PORT_PREFIX}%02[1]d"
+grpcurl -plaintext "localhost:${CANTON_PARTICIPANT_GRPC_HEALTHCHECK_PORT_PREFIX}%02[1]d" grpc.health.v1.Health/Check
+`, i)
```

Comment:
Does this Sprintf work as expected? I see 3 format verbs (%02[1]d) but just one arg in the varargs section of the args (i rather than i,i,i)

Reply:
Oh nice! TIL. 😄